packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1418,6 +1424,14 @@ class FocusManager with DiagnosticableTreeMixin {
   }
 }
 
+/// Provides convenient access to the current [FocusManager] from the
+/// [WidgetsBinding] instance.
+FocusManager get focusManager => WidgetsBinding.instance.focusManager;
```

Comment:
Looks good. Did you consider using a static method on `FocusManager` instead of a top-level getter here? Do we have other top-level globals like this? Is there any concern about name clashing with existing app code for either of these?

Reply:
Well, I did consider a static, but it would have had to have a different name (because `primaryFocus` is taken), and it's pretty useful at the top level.  There are other top level globals like these: `imageCache`, and `defaultTargetPlatform` to name a couple. Like any new name, It could clash, of course, but I don't think it's very likely to clash.

However, I think I will convert the `focusManager` one to a static "FocusManager.instance" so that it appears as the singleton that it is (but still get it internally from the `WidgetsBinding.instance.focusManager`). That way it's one less thing to clash.

Reply:
If we already have a precedence for top level names, this is probably not a big deal. Pragmatically speaking I doubt there would be much collision. I prefer not using global namespaces, but this case maybe useful enough to do so.

Or given that you already changed it to `FocusManager.instance` while I was typing this :), never-mind.